docs/messaging/messaging-spans.md

```diff
@@ -153,36 +153,35 @@ in such a way that it cannot be changed by intermediaries.
 
 ### Span name
 
-The span name SHOULD be set to the message destination name and the name of the operation being performed (as captured in [`messaging.operation.name`](../attributes-registry/messaging.md)) in the following format:
+Messaging spans SHOULD follow the overall [guidelines for span names](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.31.0/specification/trace/api.md#span).
 
-```
-<destination name> <operation name>
-```
+<!-- markdown-link-check-disable -->
+<!-- HTML anchors are not supported https://github.com/tcort/markdown-link-check/issues/225-->
+The **span name** SHOULD be `{messaging.operation.name} {destination}` (see below for the exact definition of the [`{destination}`](#destination-placeholder) placeholder).
+<!-- markdown-link-check-enable -->
 
-If the operation name is not specified by the messaging system, then the operation type as defined in [Operation types](#operation-types) SHOULD be used:
+If neither `{messaging.operation.name}` nor `{destination}` are available, span name SHOULD be `{messaging.system}`.
 
-```
-<destination name> <operation type>
-```
+Semantic conventions for individual messaging systems MAY specify different span name format and then MUST document it in [semantic conventions for specific messaging technologies](#semantic-conventions-for-specific-messaging-technologies).
 
-The destination name SHOULD only be used for the span name if it is known to be of low cardinality (cf. [general span name guidelines](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.33.0/specification/trace/api.md#span)).
-This can be assumed if it is statically derived from application code or configuration.
-Wherever possible, the real destination names after resolving logical or aliased names SHOULD be used.
-If the destination name is dynamic, such as a [conversation ID](#conversations) or a value obtained from a `Reply-To` header, it SHOULD NOT be used for the span name.
-In these cases, an artificial destination name that best expresses the destination, or a generic, static fallback like `"(anonymous)"` for [anonymous destinations](#temporary-and-anonymous-destinations) SHOULD be used instead.
+The <span id="destination-placeholder">`{destination}`</span> SHOULD describe the entity that the operation is performed against
+and SHOULD adhere to one of the following values, provided they are accessible:
 
-Examples:
+- `messaging.destination.template` or `messaging.destination.name` SHOULD be used for operations on messages or messaging destinations.
+  The `messaging.destination.template` SHOULD be used whenever it's available. Otherwise, `messaging.destination.name` SHOULD be used only when
+  the destination is known to be neither [temporary nor anonymous](#temporary-and-anonymous-destinations).
+- `server.address:server.port` SHOULD be used for other operations not targeting any specific destination(s).
```

Comment:
```suggestion
1. `messaging.destination.template` SHOULD be used when it is available.
2. `messaging.destination.name` SHOULD be used when the destination is known to be neither [temporary nor anonymous](#temporary-and-anonymous-destinations).
3. `server.address:server.port` SHOULD be used for other operations not targeting any specific destination(s).
4. Otherwise, no destination SHOULD be specified.
```

Reply:
rephrased a bit, thanks!